onnxruntime/core/mlas/lib/qgemm.cpp

```diff
@@ -210,10 +221,8 @@ MlasDynamicQGemmBatch (
     MLAS_THREADPOOL* ThreadPool
 ) {
 #if defined(USE_KLEIDIAI) && !defined(_MSC_VER)
-    //No fallback and putting in guards. This implementation is SME2 specific.
-    if(ArmKleidiAI::UseSME2){
```

Comment:
removing `if (SME2)` here because the expectation is that this will only be called if `MlasIsDynamicQGemmAvailable()` returns true.